File: 前后端/hexaing_studio_front/src/utils/api/material.ts
import request from '../request';
import type { ApiResponse } from '../types';

/**
 * 资料分类接口
 */
export interface MaterialCategory {
  id: number;
  name: string;
  orderId: number;
}

/**
 * 资料列表项接口
 */
export interface Material {
  id: number;
  fileName: string;
  fileType: string;
  fileSize: number;
  url: string;
  description: string;
  categoryId: number;
  name: string; // 分类名称
  uploadTime: string;
  uploaderId: number;
  uploader: string;
  downloadCount: number;
  isPublic: number;
  status: number;
}

/**
 * 资料详情接口
 */
export interface MaterialDetail {
  id: number;
  fileName: string;
  fileType: string;
  fileSize: number;
  url: string;
  description: string;
  categoryId: number;
  category: string;
  uploadTime: string;
  uploaderId: number;
  uploader: string;
  downloadCount: number;
  isPublic: number;
  status: number;
}

/**
 * 分页查询参数接口
 */
export interface MaterialQueryParams {
  page: number;
  pageSize: number;
  name?: string;
  categoryId?: number;
  fileType?: string;
  fileTypes?: string[];
  startDate?: string;
  endDate?: string;
  isPublic?: number;
}

/**
 * 获取资料分类列表
 */
export function getMaterialCategories(): Promise<ApiResponse<MaterialCategory[]>> {
  return request({
    url: '/admin/material/categories',
    method: 'get'
  });
}

/**
 * 获取资料列表
 * @param params 查询参数
 */
export function getMaterialList(params: MaterialQueryParams): Promise<ApiResponse<{
  total: number;
  records: Material[];
  pages: number;
}>> {
  return request({
    url: '/admin/material/list',
    method: 'post',
    data: params
  });
}

/**
 * 获取资料详情
 * @param id 资料ID
 */
export function getMaterialDetail(id: number): Promise<ApiResponse<MaterialDetail>> {
  return request({
    url: '/admin/material/detail',
    method: 'get',
    params: { id }
  });
}

/**
 * 上传资料
 * @param data 表单数据
 */
export function uploadMaterial(data: FormData): Promise<ApiResponse<object>> {
  return request({
    url: '/admin/material/upload',
    method: 'post',
    headers: {
      'Content-Type': 'multipart/form-data'
    },
    data
  });
}

/**
 * 更新资料
 * @param data 资料数据
 */
export function updateMaterial(data: {
  id: number;
  description: string;
  categoryId: number;
  isPublic: number;
}): Promise<ApiResponse<object>> {
  return request({
    url: '/admin/material/update',
    method: 'post',
    data
  });
}

/**
 * 删除资料
 * @param id 资料ID
 */
export function deleteMaterial(id: number): Promise<ApiResponse<object>> {
  return request({
    url: '/admin/material/delete',
    method: 'post',
    data: { id }
  });
}

/**
 * 记录资料下载
 * @param id 资料ID
 */
export function recordMaterialDownload(id: number): Promise<ApiResponse<object>> {
  return request({
    url: '/admin/material/download/record',
    method: 'post',
    data: { id }
  });
}

/**
 * 添加资料分类
 * @param data 分类数据
 */
export function addMaterialCategory(data: {
  name: string;
  orderId: number;
}): Promise<ApiResponse<object>> {
  return request({
    url: '/admin/material/category/add',
    method: 'post',
    data
  });
}

/**
 * 更新资料分类
 * @param data 分类数据
 */
export function updateMaterialCategory(data: {
  id: number;
  name: string;
  orderId: number;
}): Promise<ApiResponse<object>> {
  return request({
    url: '/admin/material/category/update',
    method: 'post',
    data
  });
}

/**
 * 删除资料分类
 * @param id 分类ID
 */
export function deleteMaterialCategory(id: number): Promise<ApiResponse<object>> {
  return request({
    url: '/admin/material/category/delete',
    method: 'post',
    data: { id }
  });
} 

File: 前后端/back_hexiang_studio_back/studio_pojo/src/main/java/com/back_hexiang_studio/entity/Material.java
package com.back_hexiang_studio.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 资料实体类
 */
@Data
@TableName("material")
public class Material {
    
    /**
     * 主键ID
     */
    @TableId(value = "id", type = IdType.AUTO)
    private Long id;
    
    /**
     * 文件名
     */
    @TableField("file_name")
    private String fileName;
    
    /**
     * 文件类型/扩展名
     */
    @TableField("file_type")
    private String fileType;
    
    /**
     * 文件大小(字节)
     */
    @TableField("file_size")
    private Long fileSize;
    
    /**
     * 资料访问路径
     */
    @TableField("url")
    private String url;
    
    /**
     * 资料描述
     */
    @TableField("description")
    private String description;
    
    /**
     * 分类ID
     */
    @TableField("category_id")
    private Long categoryId;
    
    /**
     * 上传时间
     */
    @TableField("upload_time")
    private LocalDateTime uploadTime;
    
    /**
     * 上传者ID
     */
    @TableField("uploader_id")
    private Long uploaderId;
    
    /**
     * 上传者姓名（非数据库字段）
     */
    @TableField(exist = false)
    private String uploaderName;
    
    /**
     * 下载次数
     */
    @TableField("download_count")
    private Long downloadCount;
    
    /**
     * 是否公开 0不公开 1公开
     */
    @TableField("is_public")
    private Integer isPublic;
    
    /**
     * 状态：1-正常，0-已删除
     */
    @TableField("status")
    private Integer status;
    
    /**
     * 更新时间
     */
    @TableField("update_time")
    private LocalDateTime updateTime;
    
    /**
     * 更新者ID
     */
    @TableField("update_id")
    private Long updateUser;
}
